concordium-consensus/src/Concordium/KonsensusV1/Consensus.hs

```diff
@@ -90,3 +90,37 @@ advanceRound newRound timedOut = do
                 when (isJust $ finalizerByBakerId (bakersAndFinalizers ^. bfFinalizers) bakerId) $
                     -- The consensus runner is a finalizer for the current epoch then we reset the timer
                     resetTimer currentTimeout
+                    
+-- |Compute and return the 'LeadershipElectionNonce' for
+-- the provided 'Epoch' and 'FinalizationEntry'
+-- TODO: implement.
+computeLeadershipElectionNonce ::
+    -- |The 'Epoch' to compute the 'LeadershipElectionNonce' for.
+    Epoch ->
+    -- |The witness for the new 'Epoch'
+    FinalizationEntry ->
+    -- |The new 'LeadershipElectionNonce'
+    LeadershipElectionNonce
+computeLeadershipElectionNonce epoch finalizationEntry = undefined
```

Comment:
This doesn't look quite right. To compute the new LE nonce you will need the temporary nonce that you update every new block and the baker set. FinalizationEntry is probably not needed. But since you haven't implemented it yet, maybe you don't care if this needs to be changed. So ignore this comment.